File: Frontend/DrivingSchoolWeb/src/app/categories/category-d/models/instructor.ts
import {Base} from 'src/app/categories/shared/models/base';
import {WorkExperience} from 'src/app/categories/category-d/models/work-experience';
import {InstructorCategoryLicence} from 'src/app/categories/category-d/models/instructor-category-licence';

export interface Instructor extends Base{
  fullname: string,
  age: number,
  licenceId: string,
  photo: string | null,
  startedDay:string,
  totalWorkExperience:number,
  totalDriveExperience:number,
  busDriveExperience:number,
  instructorAcademy:string,
  isBooked:boolean,
  categoryId:number,
  workExperiences:WorkExperience[],
  categoryLicences:InstructorCategoryLicence[]
}

File: Services/CategoryServices/DrivingSchoolApp.Services.CategoryServices.DCategoryAPI/DataAccess/Entities/Instructor.cs
namespace DrivingSchoolApp.Services.CategoryServices.DCategoryAPI.DataAccess.Entities;

public class Instructor : Base
{
	public string Firstname { get; set; }
	public string Lastname { get; set; }
	public int Age { get; set; }
	public string LicenceId { get; set; }
	public string Photo { get; set; }
	public string StartedDay { get; set; }
	public int TotalWorkExperience { get; set; }
	public int TotalDriveExperience { get; set; }
	public int BusDriveExperience { get; set; }
	public string InstructorAcademy { get; set; }
	public bool isBooked { get; set; }
	public int CategoryDId { get; set; }
	public virtual CategoryD CategoryD { get; set; }
	public IEnumerable<WorkExperience> WorkExperiencePerCompany { get; set; } = new List<WorkExperience>();
	public IEnumerable<InstructorCategoryLicence> CategoryLicences { get; set; } = new List<InstructorCategoryLicence>();
}